### DISCUSSION TITLE
Cannot connect to Home Assistant instances over "http://".

### DISCUSSION THREAD

**Discussion by eivihol** [shannonhochkins/ha-component-kit#126] [Q&A] [answered]
I am having trouble after the first step:

npm create hakit@latest

Where I get a message that I cannot connect over http://
When I copy the hassTokens to localStorage I do connect, however that will not work on every device I'm planning to use this.

Is this an issue in my HAOS?

**Comment by shannonhochkins** [comment]
Have you had a look at the options for HassConnect? It has an option to allow non secure applications  🙃

**Reply by eivihol** [reply]
Thank you very much for the quick answer! I might be very unexperienced here, but I cannot find any information regarding HassConnect. Is it in HA or do i make a config file for your package?

**Reply by shannonhochkins** [reply]
What have you currently got in your application source code

**Reply by eivihol** [reply]
The only thing I have done is:

npm create hakit@latest

and replaced the variables in .env (I have confirmed that I can ssh root@10.10.1.6 and establish a connection)

I have spent some time trying to find the settings, but I guess I'm just not looking the right place

**Reply by shannonhochkins** [reply, marked as answer]
If you're using a local server you need to enable non secure links through HassConnect. All these properties are available through typescript, however all you need to do is add the option to HassConnect

```
<HassConnect
  hassUrl={YOUR_HASS_URL}
  options={{
      allowNonSecure: true
  }}
>
  <p>
    Successfully Authenticated!
  </p>
</HassConnect>
```

Just an FYI the new version that should be released next week or so will be deprecating this property and it will allow it by default.

**Reply by eivihol** [reply]
Thank you so much! Very straight forwards. By the way, in the documentation of TimeCard, I did not see a way of formatting 24hr. Should I make a request for that?

I would also love for the sync to work, but then I need https locally, right?

**Reply by shannonhochkins** [reply]
I don't do any formatting to time sensors, it's simply how you set up your time sensor within home assistant - https://www.home-assistant.io/integrations/time_date

At this point I believe the sync is only possible with a publicly accessible URL, it's something i'll make note of to see if it's possible!
